app/src/main/java/com/example/sharewhatyoucanproject/Adapters/PostAdapter.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.sharewhatyoucanproject.Adapters
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import coil.load
+import coil.transform.RoundedCornersTransformation
+import com.example.sharewhatyoucanproject.R
+import com.example.sharewhatyoucanproject.models.PostModel
+
+class PostAdapter(
+    private val context: Context,
+    private val onPostClicked: (PostModel) -> Unit,
+) : RecyclerView.Adapter<PostAdapter.PostsViewHolder>() {
+
+    private val arrayList = mutableListOf<PostModel>()
+
+    fun setPostsList(list: List<PostModel>) {
+        arrayList.clear()
+        arrayList.addAll(list)
+        notifyDataSetChanged()
+
+    }
+
+    inner class PostsViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val posttitle: TextView = itemView.findViewById(R.id.posttitle)
+        val postdesc: TextView = itemView.findViewById(R.id.postdesc)
+        val postimg: ImageView = itemView.findViewById(R.id.postimg)
+
+
+        fun bind(postModel: PostModel) {
+            posttitle.text = postModel.title
+            postdesc.text = postModel.desc
+
+            postimg.load(postModel.image)
```

Comment:
it looks like you are calling `postimg.load` twice. Here and also in `onBindViewHolder`. I think moving the logic you have in `onBindViewHolder` to this method would make the most sense

Reply:
-changed